From 71dfedb2056e792affb9ad748541c7ef3730d90a Mon Sep 17 00:00:00 2001
From: ItsLynix <lynixthewolf@gmail.com>
Date: Wed, 4 Sep 2024 07:44:19 -0400
Subject: [PATCH 26/26] Revert "Add (partial, cam id is hardcoded) support for
 Asus ZF6 motor camera"

This reverts commit 696decb6f20605c5ab9b68d43e9595de9bc5a5cd.
---
 .../camera/libcameraservice/CameraService.cpp | 26 -------------------
 .../camera/libcameraservice/CameraService.h   |  3 ---
 2 files changed, 29 deletions(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index e2193c6..acc3b32 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -28,7 +28,6 @@
 #include <sstream>
 #include <string>
 #include <sys/types.h>
-#include <sys/wait.h>
 #include <inttypes.h>
 #include <pthread.h>
 #include <poll.h>
@@ -174,7 +173,6 @@ static std::set<std::string> sServiceErrorEventSet;
 
 CameraService::CameraService(
         std::shared_ptr<CameraServiceProxyWrapper> cameraServiceProxyWrapper,
-        mPhysicalFrontCamStatus(false),
         std::shared_ptr<AttributionAndPermissionUtils> attributionAndPermissionUtils) :
         AttributionAndPermissionUtilsEncapsulator(attributionAndPermissionUtils == nullptr ?
                 std::make_shared<AttributionAndPermissionUtils>()\
@@ -2700,7 +2698,6 @@ Status CameraService::connectHelper(const sp<CALLBACK>& cameraCb, const std::str
             mServiceLock.lock();
         } else {
             // Otherwise, add client to active clients list
-            physicalFrontCam(cameraId == "1");
             finishConnectLocked(client, partial, oomScoreOffset, systemNativeClient);
         }
 
@@ -2827,27 +2824,6 @@ status_t CameraService::addOfflineClient(const std::string &cameraId,
     return OK;
 }
 
-void CameraService::physicalFrontCam(bool on) {
-    if(on == mPhysicalFrontCamStatus) return;
-    mPhysicalFrontCamStatus = on;
-
-    if(access("/dev/asusMotoDrv", F_OK) == 0) {
-        int pid = fork();
-        if(pid == 0) {
-            const char* cmd[] = {
-                "/system/bin/asus-motor",
-                "0",
-                NULL
-            };
-            cmd[1] = on ? "0" : "1";
-            execve("/system/bin/asus-motor", (char**)cmd, environ);
-            _exit(1);
-        } else {
-            waitpid(pid, NULL, 0);
-        }
-    }
-}
-
 Status CameraService::turnOnTorchWithStrengthLevel(const std::string& unresolvedCameraId,
         int32_t torchStrength, const sp<IBinder>& clientBinder, int32_t deviceId,
         int32_t devicePolicy) {
@@ -4179,8 +4155,6 @@ binder::Status CameraService::BasicClient::disconnect() {
     }
     mDisconnected = true;
 
-    sCameraService->physicalFrontCam(false);
-
     sCameraService->removeByClient(this);
     sCameraService->logDisconnected(mCameraIdStr, mClientPid, mClientPackageName);
     sCameraService->mCameraProviderManager->removeRef(CameraProviderManager::DeviceMode::CAMERA,
diff --git a/services/camera/libcameraservice/CameraService.h b/services/camera/libcameraservice/CameraService.h
index 52762e4..9998fb8 100644
--- a/services/camera/libcameraservice/CameraService.h
+++ b/services/camera/libcameraservice/CameraService.h
@@ -285,9 +285,6 @@ public:
     // Register an offline client for a given active camera id
     status_t addOfflineClient(const std::string &cameraId, sp<BasicClient> offlineClient);
 
-    bool mPhysicalFrontCamStatus;
-    void physicalFrontCam(bool on);
-
     /////////////////////////////////////////////////////////////////////
     // Client functionality
 
-- 
2.39.2

